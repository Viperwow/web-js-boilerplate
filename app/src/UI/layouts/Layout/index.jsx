// Vendors
import React, {Component} from 'react';
import {BrowserRouter as Router, Redirect, Route, Switch} from 'react-router-dom';
// UI
import MainPage from 'src/UI/pages/MainPage';
import SecondPage from 'src/UI/pages/SecondPage';
import ThirdPage from 'src/UI/pages/ThirdPage';
import UnknownRoutePage from 'src/UI/pages/UnknownRoutePage';

class Layout extends Component {
  render() {
    return (
      <Router>
        <div className="layout">
          <Switch>
            <Redirect exact from="/" to="/1" />
            <Route exact path="/1" component={MainPage} />
            <Route exact path="/2" component={SecondPage} />
            <Route exact path="/3" component={ThirdPage} />
            <Route component={UnknownRoutePage} />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default Layout;
