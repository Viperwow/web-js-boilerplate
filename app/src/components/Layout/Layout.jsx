import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Layout extends Component {
  render() {
    const { children } = this.props;

    return (
      <div>
        {children}
      </div>
    );
  }
}

Layout.defaultProps = {
  children: [],
};

Layout.propTypes = {
  children: PropTypes.node,
};
