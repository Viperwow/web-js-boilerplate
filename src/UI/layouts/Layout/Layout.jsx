import React from 'react';
import {
  Redirect,
  Route,
  Switch,
} from 'react-router-dom';
import DynamicMainPage from 'src/UI/pages/MainPage';
import DynamicSecondPage from 'src/UI/pages/SecondPage';
import DynamicThirdPage from 'src/UI/pages/ThirdPage';
import DynamicUnknownRoutePage from 'src/UI/pages/UnknownPage';

const Layout = () => (
  <div className="layout">
    <Switch>
      <Redirect exact from="/" to="/1" />
      <Route exact path="/1" render={DynamicMainPage} />
      <Route exact path="/2" render={DynamicSecondPage} />
      <Route exact path="/3" render={DynamicThirdPage} />
      <Route exact path="/not-found" render={DynamicUnknownRoutePage} />
      <Redirect to="/not-found" />
    </Switch>
  </div>
);

export default Layout;
