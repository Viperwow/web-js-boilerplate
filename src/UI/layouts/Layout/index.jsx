import React, {Component} from 'react';
import {
  Redirect,
  Route,
  Router,
  Switch,
} from 'react-router-dom';
import DynamicMainPage from 'src/UI/pages/MainPage/DynamicMainPage';
import DynamicSecondPage from 'src/UI/pages/SecondPage/DynamicSecondPage';
import DynamicThirdPage from 'src/UI/pages/ThirdPage/DynamicThirdPage';
import DynamicUnknownRoutePage from 'src/UI/pages/UnknownRoutePage/DynamicUnknownRoutePage';
import {getHistory} from 'src/helpers/history';

class Layout extends Component {
  render() {
    return (
      <Router history={getHistory()}>
        <div className="layout">
          <Switch>
            <Redirect exact from="/" to="/1" />
            <Route exact path="/1" render={DynamicMainPage} />
            <Route exact path="/2" render={DynamicSecondPage} />
            <Route exact path="/3" render={DynamicThirdPage} />
            <Route exact path="/not-found" render={DynamicUnknownRoutePage} />
            <Redirect to="/not-found" />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default Layout;
