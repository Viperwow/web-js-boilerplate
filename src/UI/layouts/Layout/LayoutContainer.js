import withOnError from 'src/common/withOnError';
import {compose} from 'src/helpers/utility';
import Layout from './Layout';

export default compose(withOnError)(Layout);
