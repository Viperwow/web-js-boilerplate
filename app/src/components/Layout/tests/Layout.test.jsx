// Vendors
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
// UI
import Layout from '../index';

describe('Layout component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<Layout />);

    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
