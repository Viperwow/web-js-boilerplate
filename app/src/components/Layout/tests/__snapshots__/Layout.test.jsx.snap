// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render correctly With default props 1`] = `<div />`;
